src/main/kotlin/com/devfloor/untitled/article/application/ArticleFacade.kt

```diff
@@ -33,32 +33,30 @@ class ArticleFacade(
     }
 
     @Transactional(readOnly = true)
-    fun showByArticleId(articleId: Long): ArticleResponse {
-        articleService.showById(articleId).let {
-            val options = articleOptionService.showAllByArticle(it)
-            val hashtags = articleHashtagService.showAllByArticle(it)
-            val favorites = articleFavoriteService.showAllByArticle(it)
+    fun showByArticleId(articleId: Long): ArticleResponse =
+        articleService.showById(articleId)
+            .let {
+                val options = articleOptionService.showAllByArticle(it)
+                val hashtags = articleHashtagService.showAllByArticle(it)
+                val favorites = articleFavoriteService.showAllByArticle(it)
 
-            return ArticleResponse(
-                options = options,
-                article = it,
-                hashtags = hashtags,
-                favorites = favorites,
-            )
-        }
-    }
+                ArticleResponse(
+                    options = options,
+                    article = it,
+                    hashtags = hashtags,
+                    favorites = favorites,
+                )
+            }
 
     @Transactional
     fun modifyByArticleId(
         articleId: Long,
         request: ArticleModifyRequest
-    ) {
-        articleService.showById(articleId).let { article ->
-            articleService.modify(article, request.title, request.content)
+    ) = articleService.modify(articleId, request.title, request.content)
+        .let { article ->
             request.hashtags.map { hashtagService.createByName(it) }
                 .let { articleHashtagService.modifyByArticle(article, it) }
             optionService.showAllByOptionType(request.options)
                 .let { articleOptionService.modifyByArticle(article, it) }
         }
-    }
 }
```

Comment:
```kotlin
    fun modifyByArticleId(
        articleId: Long,
        request: ArticleModifyRequest
    ) {
        val article = articleService.modify(articleId, request)

        request.hashtags
            .map { hashtagService.createByName(it) }
            .let { articleHashtagService.modifyByArticle(article, it) }

        optionService.showAllByOptionType(request.options)
            .let { articleOptionService.modifyByArticle(article, it) }
    }
```

여기 코드를 이런 식으로 수정하는 건 어때? 이건 그냥 제안정도?

앞서서 다른 리뷰에서 형이
> Kotlin적으로 관용적으로 사용하라고 한 것중 하나가 Scope Function이라고 적혀있었다

라고 말해줬었는데 [kotlin 공식 문서](https://kotlinlang.org/docs/scope-functions.html#function-selection)를 읽다보니 
> Although the scope functions are a way of making the code more concise, avoid overusing them: it can decrease your code readability and lead to errors. Avoid nesting scope functions and be careful when chaining them: it's easy to get confused about the current context object and the value of `this` or `it`.

이런 식으로 설명을 해뒀더라고.

나도 위 코드를 봤을 때 코드를 이해하는데 조금 시간이 걸리더라고. 그리고 반환할 값도 없으니까 표현식으로 안쓰고 이렇게 블록 바디로 써도 읽는데 무리 없을 것 같구. 사이에 공백들로 코드 내의 문맥을 파악하기 좋게 끊어주고. 코드 내에서 문맥이 다양할 경우 이렇게 끊어서 작성하는걸 좀 더 선호하는데 한 번 참고해주면 좋을 것 같아!

Reply:
흠 아 그러게.. nesting scope 하나 더 만들어주는게 오히려 더 가독성에서 안좋은거같다.
정말 사용할 부분에서만 사용하고 너무 항상 사용하진 않는게 좋은거같네